gitops/rings/README.md

```diff
@@ -0,0 +1,144 @@
+
+For every independent service we assume 2 git repositories exist:
+
+**Service Source Repository**: A git repository that maintains the source code of the service, a dockerfile, and a helm chart. Developers will commit regularly to this repository, with revisions and rings being tracked in Git branches.
+
+**Service HLD Repository**: A second git repository that maintains a High Level Definition for the source repository. Commits to this repository are automated, and configuration is performed via AzDo pipelines. Subcomponents in this repository map to the branches in the Service Source Repository
+
+For all services represented by the above 2 git repositories, we assume two more repositories exist:
+
+**“Flatpack” HLD Repository**: A git repository that maintains a High Level Definition for all Services and Revisions that are intended to be run on the Cluster.
+
+**“Materialized” Manifest Repository**: this git repository acts as our canonical source of truth for Flux – the in-cluster component that pulls and applies Kubernetes manifests rendered from the Flatpack HLD repository.
+
+### Ring.yaml
+As a ring is considered to be strictly a revision of a microservice, we need a way to configure the ingress controller to route to the microservice revision a user belongs to. We achieve this by providing a `ring.yaml` file in our helm chart, which is an abstraction on Kubernetes and Traefik primitives.
+
+An example of ring.yaml:
+
+```
+apiVersion: apps/v1
+kind: Ring
+metadata:
+ name: myBranch
+ deployable: true or false
+spec:
+ # Turnstile data (Custom Claims)
+ claim: edge-users
+ entryPoints:
+ # Source of traffic (eg: 80 or 443)
+ - web
+ - web-secured
+ routes:
+ - PathPrefix(`/query/v1`) && Headers(`x-ring`, `myBranch`)
+ selector:
+   # Target deployment instances
+   # name of microservice
+   name: publish
+   # Major version of microservice
+   version: v1
+   # Branch name
+   ring: myBranch
+```
+
+The `ring.yaml` pairs with a `deployment.yaml` within the helm chart of your `src` repository as follows:
+
+**Source Repository**:
+ * chart
+   * templates
+       * ring.yaml
+       * deployment.yaml
+   * Chart.yaml
+   * values.yaml
+ * config
+   * common.yaml
+ * src
+   * Dockerfile
+   * ..
+ * azure-pipelines.yaml
+
+The `selector` field of your `ring.yaml` should match the `spec.selector.matchLabels` of the `deployment.yaml` file. This is because the Ring Operator will create the service that matches the labels on the `deployment.yaml`, and once the Ring is created, it knows to map to a specific version of a service and ingress routes based on these `selector` labels, and will control access control to the service.
+
+### Ring Operator
```

Comment:
We should probably add a stub or something noting that this is currently BYO (unless we plan on open sourcing the ring operator for traefik as well)

Reply:
As in BYO ring operator... or?